### ISSUE TITLE
New feature: Ignore URL Patterns

### ISSUE THREAD

**Issue by rogercyyu** [slaterslater/urlStatus#8]
Hi, I would like add a new feature to your project:
- Ignore URL Patterns (lab4)

When a user uses `--ignore` flag with a ignore file, your program will ensure those URLs are not checked if present in your URLs file.

Thanks!

**Comment by slaterslater** [comment]
Please proceed and let me know if you have any questions on how the program works. Thanks.

**Comment by slaterslater** [comment]
There is a bug inside the `strip_ignored()` function
```python
 for url in source_list:
    pathArray = url.split('/')
    url = pathArray[0] + "//" + pathArray[2]
    if url in ignore_list[0:len(url)]:
      continue
    new_source_list.append(url)
```
If a url's domain is not on the ignore list, the url's domain gets appended to `new_source_list`.
For example:
```
https://www.python.org/
https://www.python.org/about/
https://www.python.org/downloads/
``` 
Get appended to `new_source_list` as:
```
https://www.python.org
https://www.python.org
https://www.python.org
```
I made a few changes in [28e13bb](https://github.com/slaterslater/gustavo/commit/28e13bb4680d84762e43d40a15d9aa4d7c3eb5b3). Take a look and let me know what you think

**Comment by slaterslater** [comment]
Fixed by [28883bd](https://github.com/slaterslater/gustavo/commit/28883bdc80345a86c63682542e6427c3049e5272)
